nearcore/src/state_sync.rs

```diff
@@ -965,11 +963,15 @@ impl StateDumper {
     // Checks which parts have already been uploaded possibly by other nodes
     // We use &mut so the do_state_sync_dump() future will be Send, which it won't be if we use a normal
     // reference because of the Chain field
-    async fn check_stored_parts(&mut self) {
+    fn check_stored_parts(&self) -> impl Future<Output = ()> {
```

Comment:
Ditto.

Reply:
The comment above this method needs to be adjusted.